### ISSUE TITLE
Not capturing traffic using six.moves.http_client.HTTPConnection

### ISSUE THREAD

**Issue by photuris** [httptoolkit/feedback#58]
This is probably just user error. But I'm stumped.

HTTP Toolkit is working fine for intercepting requests using requests and the like.

But I'm using a third party library (splunklib) that relies on six.moves.http_client, and for some reason, none of that traffic is being captured. I have no idea why.

Any ideas?

Thank you!

Here's a snip of code that handles requests:

``` python
    def request(url, message, **kwargs):
        scheme, host, port, path = _spliturl(url)
        body = message.get("body", "")
        head = {
            "Content-Length": str(len(body)),
            "Host": host,
            "User-Agent": "splunk-sdk-python/1.6.9",
            "Accept": "*/*",
            "Connection": "Close",
        } # defaults
        for key, value in message["headers"]:
            head[key] = value
        method = message.get("method", "GET")

        connection = connect(scheme, host, port)
        is_keepalive = False
        try:
            connection.request(method, path, body, head)
            if timeout is not None:
                connection.sock.settimeout(timeout)
            response = connection.getresponse()
            is_keepalive = "keep-alive" in response.getheader("connection", default="close").lower()
        finally:
            if not is_keepalive:
                connection.close()

        return {
            "status": response.status,
            "reason": response.reason,
            "headers": response.getheaders(),
            "body": ResponseReader(response, connection if is_keepalive else None),
        }
```

---

_Does this affect you too? Click the +:smiley: icon in the top right and add a :+1: on issues to help decide where HTTP Toolkit goes next, or [go take a look & vote on the most popular ideas so far](https://github.com/httptoolkit/feedback/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc)._

**Comment by pimterry** [comment]
Hi @photuris. I suspect that six is internally using an HTTP client that isn't automatically configured correctly. I'll investigate further and see what I can come up with. What Python version are you using?

**Comment by photuris** [comment]
Python 2.7 (I know!).

As for not being configured correctly, how might one configure it correctly?

I'll do some local patching if that's what it takes.

Thank you!

**Comment by photuris** [comment]
Ok, so far, six.moves.urllib seems to work ok, but six.moves.http_client does not (which is what the library we depend on uses).

I'll keep messing with it.

**Comment by pimterry** [comment]
Yes, http_client wasn't working. I have a patch available, it's not released yet but you're welcome to test it out.

You can do so by copying the contents of https://github.com/httptoolkit/httptoolkit-server/tree/master/overrides/pythonpath onto your machine. The local path you need is listed in `$PYTHONPATH` env in the intercepted terminal. Let me know if it works for you!

Full release should be out within a few days, if I don't find any issues in testing.

**Comment by photuris** [comment]
YES!!!!! It works!

Thank you!

I can knock the rest of this out today. You're awesome!

**Comment by pimterry** [comment]
Haha, great! Glad that's sorted it for you. I'll update & close this ticket once this has been released officially, shouldn't be long.

**Comment by pimterry** [comment]
Just to let you know, this has now been officially released in server 0.1.29. Your app will update automatically, so this shouldn't appear in future. Thanks for the report, glad we could get this fixed!
